internal/v2/service/stats.go

```diff
@@ -77,16 +88,59 @@ func (s *V2Service) GetOverallStats(
 		phase1Stats.ActiveDelegations = 0
 	}
 
+	btcStakingAPY, errApyCalculation := s.GetBTCStakingAPY(ctx, phase1Stats.ActiveTvl)
+	if errApyCalculation != nil {
+		log.Ctx(ctx).Error().Err(errApyCalculation).
+			Msg("error while calculating BTC staking APY")
+		return nil, types.NewInternalServiceError(errApyCalculation)
+	}
+
 	return &OverallStatsPublic{
 		ActiveTvl:               overallStats.ActiveTvl,
 		ActiveDelegations:       overallStats.ActiveDelegations,
 		TotalActiveTvl:          overallStats.ActiveTvl + phase1Stats.ActiveTvl,
 		TotalActiveDelegations:  overallStats.ActiveDelegations + phase1Stats.ActiveDelegations,
 		ActiveFinalityProviders: uint64(activeFinalityProvidersCount),
 		TotalFinalityProviders:  uint64(len(finalityProviders)),
+		BTCStakingAPY:           btcStakingAPY,
 	}, nil
 }
 
+func (s *V2Service) GetBTCStakingAPY(
+	ctx context.Context, activeTvl int64,
+) (float64, *types.Error) {
+	// Skip calculation if activeTvl is 0
+	if activeTvl <= 0 {
+		return 0, nil
+	}
+
+	// CoinMarketCap integration is optional since not all deployments require APY calculation.
+	// If CoinMarketCap is not configured in the service config, return 0 as the APY.
+	if s.clients.CoinMarketCap == nil {
+		return 0, nil
+	}
+
+	// Convert the activeTvl to BTC as APY is calculated per BTC
+	btcTvl := float64(activeTvl) / 1e8
+
+	btcPrice, err := s.sharedService.GetLatestBTCPrice(ctx)
+	if err != nil {
+		return 0, types.NewInternalServiceError(err)
+	}
+
+	babyPrice, err := s.sharedService.GetLatestBABYPrice(ctx)
+	if err != nil {
+		return 0, types.NewInternalServiceError(err)
+	}
+
+	// Calculate the APY of the BTC staking on Babylon Genesis
+	// APY = (400,000,000 * BABY/BTC price) / Total BTC staked * 100
+	btcStakingAPY := (ANNUAL_BABY_REWARDS_FOR_BTC_STAKING * (babyPrice / btcPrice)) / btcTvl * 100
+
+	// Round the APY to 2 decimal places
+	return math.Round(btcStakingAPY*100) / 100, nil
```

Comment:
I would leave more decimal places and let the clients decide how they want to display.